pallets/order-book/src/lib.rs

```diff
@@ -488,6 +554,8 @@ pub mod pallet {
             let dex_id = order_book.dex_id;
 
             order_book.cancel_limit_order::<Self>(order, &mut data)?;
+            Self::unschedule(expires_at, order_book_id, order_id)
+                .map_err(|e| <UnscheduleError as Into<Error<T>>>::into(e))?;
```

Comment:
Unscheduling should be move in `OrderBook` because it should be unscheduled when order is canceled or executed

Reply:
Btw, I would move `schedule()` in `OrderBook` as well to be consistent